analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -118,10 +118,13 @@ private static void NamedSymbolAction(SonarSymbolReportingContext context, HashS
             foreach (var declaration in namedType.DeclaringSyntaxReferences.Where(r => !r.SyntaxTree.IsGenerated(CSharpGeneratedCodeRecognizer.Instance, compilation))
                                                                            .SelectMany(x => x.GetSyntax().ChildNodes().OfType<BaseTypeDeclarationSyntax>()))
             {
-                var semanticModel = compilation.GetSemanticModel(declaration.SyntaxTree);
-                var declarationSymbol = semanticModel.GetDeclaredSymbol(declaration);
-                var symbolsCollector = RetrieveRemovableSymbols(declarationSymbol, compilation);
-                CopyRetrievedSymbols(symbolsCollector, privateSymbols, internalSymbols, fieldLikeSymbols);
+                if (compilation.GetSemanticModel(declaration.SyntaxTree) is { } semanticModel
+                    && semanticModel.GetDeclaredSymbol(declaration) is { } declarationSymbol
```

Comment:
I think we can remove the null checks
```suggestion
                if (compilation.GetSemanticModel(declaration.SyntaxTree) is var semanticModel
                    && semanticModel.GetDeclaredSymbol(declaration) is var declarationSymbol
```